library/Booster/JsonRpc.hs

```diff
@@ -512,10 +500,10 @@ mkLogRewriteTrace
                                 , origin = Booster
                                 }
             RewriteSimplified equationTraces Nothing
-                | fromMaybe False logSuccessfulSimplifications || fromMaybe False logFailedSimplifications ->
-                    mapM (mkLogEquationTrace equationLogOpts) equationTraces
+                | logSuccessfulSimplifications || logFailedSimplifications -> mapM (mkLogEquationTrace equationLogOpts) equationTraces
+                | otherwise -> Just []
```

Comment:
This case was missing and the call of `mkLogRewriteTrace` on a `RewriteSimplified` returned `Nothing` if `logSuccessfulSimplifications || logFailedSimplifications` was `False`. In other words, some rewriting traces were filtered-out if simplification logs were not requested.